Modules/_cryptmodule.c

```diff
@@ -4,6 +4,9 @@
 #include "Python.h"
 
 #include <sys/types.h>
+#ifdef HAVE_CRYPT_H
+#include <crypt.h>
```

Comment:
Do we need to `#define _GNU_SOURCE` here?

Reply:
I _think_ no because it's defined by pyconfig.h which is included by Python.h. I'm going to try a build on my Linux box to confirm.

Reply:
I compiled Python 3.7 on my Debian 9 box with this patch (because that was easiest), and `crypt.crypt()` works without segfaulting and links `crypt_r` and there are no warnings in the build log. From inspection, pyconfig.h does define _GNU_SOURCE and does get pulled in by Python.h.

Reply:
Okay, great.